resources/helm/v2.1/istio-control/istio-discovery/files/injection-template.yaml

```diff
@@ -117,8 +147,8 @@ template: |
     - sysctl -w kernel.core_pattern=/var/lib/istio/data/core.proxy && ulimit -c unlimited
     command:
       - /bin/sh
-  {{- if contains "/" .Values.global.proxy_init.image }}
-    image: "{{ .Values.global.proxy_init.image }}"
+  {{- if contains "/" (annotation .ObjectMeta `sidecar.istio.io/proxyImage` .Values.global.proxy_init.image) }}
+    image: "{{ annotation .ObjectMeta `sidecar.istio.io/proxyImage` .Values.global.proxy_init.image }}"
```

Comment:
this change looks wrong

Reply:
again, I think it's fine, see above. it just adds an indirection, it'll look at the annotation prior to the global value